solidity/contracts/AltheaDB.sol

```diff
@@ -1,54 +1,369 @@
 //SPDX-License-Identifier: Apache-2.0
 pragma solidity 0.8.21; // Force solidity compliance
 
+/// Identity struct
+struct Identity {
+    uint128 mesh_ip;
+    uint256 wg_key;
+    address eth_addr;
+}
+
+/// Thrown when the caller is not authorized to register users
+error UnathorizedCaller();
+error DuplicateUser();
+error DuplicateAdmin();
+error IdentityNotFound();
+error AdminNotFound();
+
 contract AltheaDB {
-    constructor() {}
+    /// The admin address that is allowed to update who is on the
+    /// user admin and exit admin lists. This could be an individual
+    /// account or a multisig
+    address public immutable state_admin;
+    /// A list of addresses allowed to add and remove users from the list of users
+    address[] public state_UserAdmins;
+    /// A list of addresses allowed to add and remove exits from the list of exits
+    address[] public state_ExitAdmins;
+
+    // Mappings to regsitered clients
+    Identity[] public state_registeredUsers;
+    // Mappings to regsitered exits
+    Identity[] public state_registeredExits;
```

Comment:
The purpose of this is to replace the cluster members list in the config of current exits. Right now exits need to have all cluster members configured manually. Since our goal is plug and go exits with almost no configuration making exit operators put in a list of cluster members goes against that goal.